### ISSUE TITLE
The React implementation of the Modal component should support role="alertdialog" 

### ISSUE THREAD

**Issue by FrivalszkyP** [carbon-design-system/carbon#5960]
## Detailed description

[WAI-ARIA 1.1 describes an alertdialog](https://www.w3.org/TR/wai-aria-1.1/#alertdialog) role. To quote: 

> Dialogs are most often used to prompt the user to enter or respond to information. **A dialog that is designed to interrupt workflow is usually modal. See related alertdialog.**

Carbon already supports the `danger={true}` prop which, in my opinion, should go hand in hand with an `alert` prop that changes the `role` of the modal from `dialog` to `alertdialog`. 

When an `alertdialog` role is set, assistive technology can announce the contents of the dialog upon being displayed, much like the contents of a `role="alert"` element are announced. This is desirable for dialogs that should interrupt workflow (see the ARIA description above).

It is also important that the WAI-ARIA guidelines state that a role="alertdialog" element should be supplied with an `aria-label` or `aria-labelledby` attribute as well as an `aria-describedby` attribute. The former is already taken care of by the Modal component, but an `aria-describedby` attribute should be added for alertdialogs that should point to the body text of the modal (i.e. the element that contains the props.children).

**Comment by dakahn** [comment]
Seems sound to me while referencing the APG here
https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal

Transferred this from a bug report to an enhancement request.

**Comment by FrivalszkyP** [comment]
Great! You can take a look at #5955 to see if it covers the issue.
